src/engine/client/text.cpp

```diff
@@ -558,7 +558,7 @@ class CTextRender : public IEngineTextRender
 				const char *pStrBegin = pStr+COLOR_CODE_LEN+3;
 
 				// color code at string end? Don't render it.
-				if(pStrBegin == '\0')
+				if(pStrBegin == NULL)
```

Comment:
Uh I believe this was supposed to be `*pStrBegin == '\0'` but someone forgot the `*`. From the looks of it, checking the pointer for NULL doesn't make much sense as it can impossibly be NULL due to the line above.

Reply:
Oops! Good catch!